tests/nat/middleware/test_dynamic_middleware.py

```
@@ -0,0 +1,717 @@
+    from nat.middleware.utils.workflow_inventory import RegisteredComponentMethod
+
+    config = DynamicMiddlewareConfig()
+    middleware = DynamicFunctionMiddleware(config=config, builder=Mock(_functions={}))
+
+    # Create a registered component method object that's not actually registered
+    fake_registered = RegisteredComponentMethod(key="fake.method",
+                                                component_instance=Mock(),
+                                                function_name="method",
+                                                original_callable=lambda: None)
+
+    with pytest.raises(ValueError, match="'fake.method' is not registered"):
```

Ruff (0.14.7)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)